lib/backend/k8s/k8s.go

```diff
@@ -418,9 +431,11 @@ func (c *KubeClient) Get(k model.Key) (*model.KVPair, error) {
 	case model.ReadyFlagKey:
 		return c.getReadyStatus(k.(model.ReadyFlagKey))
 	case model.IPPoolKey:
-		return c.ipPoolClient.Get(k.(model.IPPoolKey))
+		return c.ipPoolClient.Get(k)
 	case model.NodeKey:
 		return c.nodeClient.Get(k.(model.NodeKey))
```

Comment:
If you like I can update the Global config as part of this PR - that way all TPR based stuff is using this approach - and the only difference then are non-TPR ones.